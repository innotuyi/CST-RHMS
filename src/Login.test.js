import React from "react";
import Login from "./components/Login";

import { configure, shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

let wrapper;

beforeEach(() => {
  wrapper = shallow(<Login />);
});

afterEach(() => {
  jest.clearAllMocks();
});

describe("Login Tests", () => {

  it("matches Snapshot Collectly", () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should have input for email and password", () => {
    expect(wrapper.find("#email")).toHaveLength(1);
    expect(wrapper.find("#password")).toHaveLength(1);
  });

  it("should have a btn component", () => {
    expect(wrapper.find("Button")).toHaveLength(1);

    expect(wrapper.find("Button").text()).toEqual("Login");
  });

  it("Should capture Email correctly onChange", () => {
    wrapper.find("#email").simulate("change", {
      target: {
        value: "admin@gmail.com",
      },
    });
    expect(wrapper.find("#email").props().value).toEqual("admin@gmail.com");
  });

  it("Should capture Password correctly onChange", () => {
    wrapper.find("#password").simulate("change", {
      target: {
        value: "12345",
      },
    });
    expect(wrapper.find("#password").props().value).toEqual("12345");
  });

  it("validates model on button click", () => {});
});
