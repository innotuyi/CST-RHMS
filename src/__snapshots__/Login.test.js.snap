// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Tests matches Snapshot Collectly 1`] = `
<div
  style={
    Object {
      "background": "linear-gradient(to bottom, white, 84%, #1089AF)",
      "height": "100vh",
    }
  }
>
  <Header />
  <Container
    fluid={false}
  >
    <Row
      className="d-flex justify-content-center arighn-item-center"
      noGutters={false}
    >
      <Col
        className="leftimage"
      >
        <Image
          fluid={false}
          rounded={false}
          roundedCircle={false}
          src="Lecturesvg.svg"
          style={
            Object {
              "height": "80%",
              "objectFit": "contain",
              "paddingTop": "80px",
              "width": "80%",
            }
          }
          thumbnail={false}
        />
      </Col>
      <Col
        className="d-flex justify-content-center"
      >
        <Jumbotron
          fluid={false}
          style={
            Object {
              "background": "linear-gradient(to bottom, white , 90%, #1089AF )",
              "borderRadius": "10px",
              "width": "100%",
            }
          }
        >
          <Form
            className="form"
            inline={false}
            onSubmit={[Function]}
            style={Object {}}
          >
            <h2
              className="text-center mb-5 text-secondary"
            >
              Login Here
            </h2>
            <FormGroup>
              <FormLabel
                column={false}
                srOnly={false}
              >
                Enter Your Email
              </FormLabel>
              <FormControl
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="Your College Email"
                required={true}
                type="email"
                value=""
              />
            </FormGroup>
            <FormGroup>
              <FormLabel
                column={false}
                srOnly={false}
              >
                Enter Your Password
              </FormLabel>
              <FormControl
                id="password"
                onChange={[Function]}
                placeholder="Your Password"
                required={true}
                type="password"
                value=""
              />
            </FormGroup>
            <Col
              className="text-center"
            >
              <Button
                active={false}
                className="lbutton"
                disabled={false}
                onClick={[Function]}
                type="submit"
                variant="primary"
              >
                Login
              </Button>
            </Col>
          </Form>
        </Jumbotron>
      </Col>
    </Row>
  </Container>
</div>
`;
